### ISSUE TITLE
Unrecognized 'cfg_branch' when using 'spm_image.m'

### ISSUE THREAD

**Issue by CenXiang** [spm/spm#27]
When attempting to display a correct T1.nii file using the spm_image function in MATLAB's SPM toolbox (spm_image('Display', fileT1);), the following error is encountered:

Error in  cfg_getfile>reg_filter (Line 1185)
    dprms  = {cfg_branch};

Error in  cfg_getfile (Line 177 )
            t = reg_filter(varargin{2:end});

Error in spm_select (Line 115)
            cfg_getfile('regfilter', 'mesh',  {'\.gii$'});

Error in spm_select (Line 110)
            spm_select('regfilter');

Error in spm_select (Line 97)
    spm_select('init');

Error in spm (第 919 行)
        fn   = cellstr(spm_select('List',tdir,['^.*' d{j} '\.m$']));

Error in spm_orthviews>reset_st (Line 1551)
xTB = spm('TBs');

Error in spm_orthviews (Line 251)
if isempty(st), reset_st; end

Error in spm_image (Line376)
    spm_orthviews('Reset');

Error in spm_image>init_display (Line 398)
spm_image('Reset');

Error in spm_image (Line 87)
    init_display(P);

Error in corregister_normalize_Images (Line 29)
spm_image('Display',fileT1);

Reinstallation of SPM toolbox did not resolve the issue.
And I re-install SPM, it can't work.  Seeking guidance on resolving this error and successfully displaying T1.nii files using the spm_image function in SPM toolbox. Thx!

**Comment by VolkmarGlauche** [comment]
My first guess is that you tried to use spm_image without initialising SPM before. Before you use any SPM function in a standalone script, run

`spm('defaults', 'FMRI');`

or whichever modality you want to use primarily. This will set up various defaults, paths and configurations that are needed to run SPM functions.

**Comment by VolkmarGlauche** [comment]
Then, you are probably missing 

`spm_jobman('initcfg');`

as a second step in your script initialisation after setting the defaults.

**Comment by CenXiang** [comment]
> Then, you are probably missing
> 
> `spm_jobman('initcfg');`
> 
> as a second step in your script initialisation after setting the defaults.

OK, thanks for your instruction, it works!

**Comment by vlitvak** [comment]
I'm pretty sure that line runs when you do:

spm('defaults','fmri');

I never had to run it separately. Might be worth figuring out what's going on even if adding it solves the immediate problem.

**Comment by CenXiang** [comment]
Yes, now I execute spm('defaults','fmri'); alone and comment out spm_jobman(); it works again, so I guess it might be an initialization issue after installing SPM?

**Comment by VolkmarGlauche** [comment]
> I'm pretty sure that line runs when you do:
> 
> spm('defaults','fmri');
> 
> I never had to run it separately. Might be worth figuring out what's going on even if adding it solves the immediate problem.

@vlitvak If you run an SPM batch after initialising the defaults, you don't need that. However, if you run something else that depends on the batch system implicitly (as e.g. spm_select does), you need to initialise it before.

@CenXiang The initialisation will persist until you restart MATLAB or do something like

`clear all`

Otherwise, make sure that there is a single SPM folder in your MATLAB path. Do not add subfolders of the SPM folder manually, SPM will take care of that.

**Comment by vlitvak** [comment]
Thanks for the explanation. Always something new to learn in SPM :-)

**Comment by VolkmarGlauche** [comment]
@vlitvak I had to read in the code as well to understand what might be going on...

**Comment by CenXiang** [comment]
Thank you for your help! I have learned valuable insights and tips on using spm, taking my first step forward 👍 :).

**Comment by CenXiang** [comment]
Got it, absolutely valuable advice, any minor problems with the program will affect the final result. Thank you both for your help.
